app/mobile-content/pages/edit-page-dialog/edit-page-dialog.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Inject, OnInit, DoCheck } from '@angular/core';
+import { MdDialogRef, MD_DIALOG_DATA } from '@angular/material';
+
+import { MobilePage, pageType } from '../shared/mobile-page.model';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hip-edit-page-dialog',
+  templateUrl: 'edit-page-dialog.component.html'
+})
+export class EditPageDialogComponent implements OnInit, DoCheck {
+  page: MobilePage;
+  private prevType: pageType;
+
+  constructor(public dialogRef: MdDialogRef<EditPageDialogComponent>,
+              @Inject(MD_DIALOG_DATA) public data: { pageToEdit: MobilePage }) {}
+
+  ngOnInit() {
+    if (this.data && this.data.pageToEdit) {
+      // deep clone input page to make editing cancelable
+      this.page = MobilePage.parseObject(JSON.parse(JSON.stringify(this.data.pageToEdit)));
```

Comment:
Hm... It feels like there should be a better way to clone an object, but `Object.assign` only does a shallow clone and I can't fight a reasonably simple solution rn. Weird.